### PR TITLE
scan: generate calldata struct and encode function in go

### LINKED ISSUE #1290: Generate calldata struct and encode function in go
Generate struct and encode function for calldata

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #1290

### CODE REVIEW

File: scan/src/bindings/Borsh.re [hunk 1]
```diff
@@ -265,3 +265,37 @@ func DecodeResult(data []byte) (Result, error) {
...
+  | U8 => {j|encoder.EncodeU8(result.$name)|j}
+  | U32 => {j|encoder.EncodeU32(result.$name)|j}
+  | U64 => {j|encoder.EncodeU64(result.$name)|j}
+  | String => {j|encoder.EncodeString(result.$name)|j}
+  };
+};
+
+let generateEncodeGo = (packageName, schema, name) => {
+  let template = (structs, functions) => {j|package $packageName
+
```
Comment (taobun) [inline scan/src/bindings/Borsh.re:281]: Add import from our library and add at decode too

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/__tests__/Borsh_test.re [hunk 2]
```diff
@@ -350,3 +354,86 @@ func DecodeResult(data []byte) (Result, error) {
...
+}
+
+func(result *Result) EncodeResult() []byte {
+	encoder := borsh.NewBorshEncoder()
+
+	encoder.EncodeString(result.symbol)
+	encoder.EncodeU64(result.multiplier)
+	encoder.EncodeU8(result.what)
+
+	return encoder.data
```
Comment (taobun) [inline scan/__tests__/Borsh_test.re]: encode.GetEncodedData

File: scan/__tests__/Borsh_test.re [hunk 3]
```diff
@@ -350,3 +354,86 @@ func DecodeResult(data []byte) (Result, error) {
...
+type Result struct {
+	Px uint64
+}
+
+func(result *Result) EncodeResult() []byte {
+	encoder := borsh.NewBorshEncoder()
+
+	encoder.EncodeU64(result.px)
+
+	return encoder.data
```
Comment (taobun) [inline scan/__tests__/Borsh_test.re]: ditto

File: scan/src/bindings/Borsh.re [hunk 4]
```diff
@@ -265,3 +267,39 @@ func DecodeResult(data []byte) (Result, error) {
...
+type Result struct {
+\t$structs
+}
+
+func(result *Result) EncodeResult() []byte {
+\tencoder := borsh.NewBorshEncoder()
+
+\t$functions
+
+\treturn encoder.data
```
Comment (taobun) [inline scan/src/bindings/Borsh.re]: root cause

**Review by taobun — CHANGES_REQUESTED** [review]

**Comment by sorawit** [comment]
@tansawit @taobun @evilpeach Let's have this merged today

**Review by evilpeach — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**